// @prisma/client/index-browser.js

/* !!! This is code generated by Prisma. Do not edit directly. !!!
/* eslint-disable */

Object.defineProperty(exports, "__esModule", { value: true });

const {
  Decimal,
  objectEnumValues,
  makeStrictEnum,
  Public,
  getRuntime,
  skip
} = require('./runtime/index-browser.js')


const Prisma = {}

exports.Prisma = Prisma
exports.$Enums = {}

/**
 * Prisma Client JS version: 6.18.0
 * Query Engine version: 34b5a692b7bd79939a9a2c3ef97d816e749cda2f
 */
Prisma.prismaVersion = {
  client: "6.18.0",
  engine: "34b5a692b7bd79939a9a2c3ef97d816e749cda2f"
}

Prisma.PrismaClientKnownRequestError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientKnownRequestError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)};
Prisma.PrismaClientUnknownRequestError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientUnknownRequestError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.PrismaClientRustPanicError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientRustPanicError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.PrismaClientInitializationError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientInitializationError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.PrismaClientValidationError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientValidationError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.Decimal = Decimal

/**
 * Re-export of sql-template-tag
 */
Prisma.sql = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`sqltag is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.empty = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`empty is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.join = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`join is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.raw = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`raw is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.validator = Public.validator

/**
* Extensions
*/
Prisma.getExtensionContext = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`Extensions.getExtensionContext is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.defineExtension = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`Extensions.defineExtension is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}

/**
 * Shorthand utilities for JSON filtering
 */
Prisma.DbNull = objectEnumValues.instances.DbNull
Prisma.JsonNull = objectEnumValues.instances.JsonNull
Prisma.AnyNull = objectEnumValues.instances.AnyNull

Prisma.NullTypes = {
  DbNull: objectEnumValues.classes.DbNull,
  JsonNull: objectEnumValues.classes.JsonNull,
  AnyNull: objectEnumValues.classes.AnyNull
}



/**
 * Enums
 */

exports.Prisma.TransactionIsolationLevel = makeStrictEnum({
  ReadUncommitted: 'ReadUncommitted',
  ReadCommitted: 'ReadCommitted',
  RepeatableRead: 'RepeatableRead',
  Serializable: 'Serializable'
});

exports.Prisma.ScenarioScalarFieldEnum = {
  id: 'id',
  code: 'code',
  name: 'name',
  noiseTolerance: 'noiseTolerance',
  spaceRequirement: 'spaceRequirement',
  iconUrl: 'iconUrl',
  isActive: 'isActive',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.EquipmentScalarFieldEnum = {
  id: 'id',
  code: 'code',
  name: 'name',
  category: 'category',
  recognizable: 'recognizable',
  recognitionLabels: 'recognitionLabels',
  recognitionConfidence: 'recognitionConfidence',
  iconUrl: 'iconUrl',
  imageUrl: 'imageUrl',
  displayOrder: 'displayOrder',
  isActive: 'isActive',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.ExerciseScalarFieldEnum = {
  id: 'id',
  code: 'code',
  name: 'name',
  primaryMuscle: 'primaryMuscle',
  secondaryMuscles: 'secondaryMuscles',
  intentType: 'intentType',
  difficulty: 'difficulty',
  description: 'description',
  defaultDuration: 'defaultDuration',
  defaultSets: 'defaultSets',
  durationType: 'durationType',
  demoImageUrl: 'demoImageUrl',
  demoVideoUrl: 'demoVideoUrl',
  tags: 'tags',
  isActive: 'isActive',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.ExerciseScenarioScalarFieldEnum = {
  exerciseId: 'exerciseId',
  scenarioId: 'scenarioId',
  createdAt: 'createdAt'
};

exports.Prisma.ExerciseEquipmentScalarFieldEnum = {
  exerciseId: 'exerciseId',
  equipmentId: 'equipmentId',
  isRequired: 'isRequired',
  createdAt: 'createdAt'
};

exports.Prisma.UserScalarFieldEnum = {
  id: 'id',
  email: 'email',
  name: 'name',
  avatarUrl: 'avatarUrl',
  totalWorkouts: 'totalWorkouts',
  totalDurationSec: 'totalDurationSec',
  currentStreak: 'currentStreak',
  longestStreak: 'longestStreak',
  preferredIntents: 'preferredIntents',
  preferredDifficulty: 'preferredDifficulty',
  preferredDuration: 'preferredDuration',
  avoidEquipment: 'avoidEquipment',
  streakReminder: 'streakReminder',
  themeWeekReminder: 'themeWeekReminder',
  quietHoursStart: 'quietHoursStart',
  quietHoursEnd: 'quietHoursEnd',
  hideRealPhotos: 'hideRealPhotos',
  autoBlurFaces: 'autoBlurFaces',
  allowDataSync: 'allowDataSync',
  language: 'language',
  theme: 'theme',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.WorkoutSessionScalarFieldEnum = {
  id: 'id',
  userId: 'userId',
  intentType: 'intentType',
  scenarioId: 'scenarioId',
  targetMuscles: 'targetMuscles',
  totalDuration: 'totalDuration',
  difficulty: 'difficulty',
  isSilent: 'isSilent',
  status: 'status',
  startedAt: 'startedAt',
  completedAt: 'completedAt',
  actualDuration: 'actualDuration',
  followMode: 'followMode',
  currentStep: 'currentStep',
  pauseCount: 'pauseCount',
  skipCount: 'skipCount',
  isOffline: 'isOffline',
  ambientNoise: 'ambientNoise',
  usedSpace: 'usedSpace',
  rating: 'rating',
  feedback: 'feedback',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.SessionExerciseScalarFieldEnum = {
  id: 'id',
  sessionId: 'sessionId',
  exerciseId: 'exerciseId',
  sequenceOrder: 'sequenceOrder',
  duration: 'duration',
  sets: 'sets',
  isCompleted: 'isCompleted',
  actualDuration: 'actualDuration',
  startedAt: 'startedAt',
  endedAt: 'endedAt',
  pausedTimes: 'pausedTimes',
  skipReason: 'skipReason',
  difficultyFelt: 'difficultyFelt',
  comfortLevel: 'comfortLevel',
  effectivenessRating: 'effectivenessRating',
  createdAt: 'createdAt'
};

exports.Prisma.ShareCardScalarFieldEnum = {
  id: 'id',
  userId: 'userId',
  sessionId: 'sessionId',
  cardImageUrl: 'cardImageUrl',
  cardTemplate: 'cardTemplate',
  cardData: 'cardData',
  rarity: 'rarity',
  equipmentSeries: 'equipmentSeries',
  rarityScore: 'rarityScore',
  dataSource: 'dataSource',
  specialTags: 'specialTags',
  cityEdition: 'cityEdition',
  themeWeek: 'themeWeek',
  shareText: 'shareText',
  isPublic: 'isPublic',
  shareCount: 'shareCount',
  viewCount: 'viewCount',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.DailyTrainingScalarFieldEnum = {
  id: 'id',
  userId: 'userId',
  trainingDate: 'trainingDate',
  totalSessions: 'totalSessions',
  totalDuration: 'totalDuration',
  totalExercises: 'totalExercises',
  completedSessions: 'completedSessions',
  intentBreakdown: 'intentBreakdown',
  muscleBreakdown: 'muscleBreakdown',
  isStreakDay: 'isStreakDay',
  achievements: 'achievements',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.RarityTableScalarFieldEnum = {
  id: 'id',
  equipmentId: 'equipmentId',
  equipmentCode: 'equipmentCode',
  weekStart: 'weekStart',
  rarityScore: 'rarityScore',
  rarityLevel: 'rarityLevel',
  dataSource: 'dataSource',
  region: 'region',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.UserPreferenceScalarFieldEnum = {
  id: 'id',
  userId: 'userId',
  preferenceType: 'preferenceType',
  preferenceKey: 'preferenceKey',
  preferenceValue: 'preferenceValue',
  usageCount: 'usageCount',
  successRate: 'successRate',
  averageRating: 'averageRating',
  lastUsedAt: 'lastUsedAt',
  firstUsedAt: 'firstUsedAt',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.ThemeWeekScalarFieldEnum = {
  id: 'id',
  title: 'title',
  code: 'code',
  description: 'description',
  equipmentCode: 'equipmentCode',
  targetExerciseCount: 'targetExerciseCount',
  startDate: 'startDate',
  endDate: 'endDate',
  rewardType: 'rewardType',
  rewardData: 'rewardData',
  status: 'status',
  isVisible: 'isVisible',
  displayOrder: 'displayOrder',
  totalParticipants: 'totalParticipants',
  totalCompletions: 'totalCompletions',
  completionRate: 'completionRate',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.ThemeWeekParticipationScalarFieldEnum = {
  id: 'id',
  userId: 'userId',
  themeWeekId: 'themeWeekId',
  status: 'status',
  joinedAt: 'joinedAt',
  completedAt: 'completedAt',
  exercisesCompleted: 'exercisesCompleted',
  targetExercises: 'targetExercises',
  progressPercent: 'progressPercent',
  rewardEarned: 'rewardEarned',
  rewardClaimedAt: 'rewardClaimedAt',
  relatedSessions: 'relatedSessions',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.DeeplinkScalarFieldEnum = {
  id: 'id',
  code: 'code',
  targetType: 'targetType',
  targetId: 'targetId',
  createdBy: 'createdBy',
  expiresAt: 'expiresAt',
  createdAt: 'createdAt'
};

exports.Prisma.DeeplinkClickScalarFieldEnum = {
  id: 'id',
  deeplinkId: 'deeplinkId',
  clickedAt: 'clickedAt',
  ipAddress: 'ipAddress',
  userAgent: 'userAgent',
  referer: 'referer'
};

exports.Prisma.SortOrder = {
  asc: 'asc',
  desc: 'desc'
};

exports.Prisma.JsonNullValueInput = {
  JsonNull: Prisma.JsonNull
};

exports.Prisma.NullableJsonNullValueInput = {
  DbNull: Prisma.DbNull,
  JsonNull: Prisma.JsonNull
};

exports.Prisma.QueryMode = {
  default: 'default',
  insensitive: 'insensitive'
};

exports.Prisma.NullsOrder = {
  first: 'first',
  last: 'last'
};

exports.Prisma.JsonNullValueFilter = {
  DbNull: Prisma.DbNull,
  JsonNull: Prisma.JsonNull,
  AnyNull: Prisma.AnyNull
};
exports.NoiseLevel = exports.$Enums.NoiseLevel = {
  SILENT: 'SILENT',
  QUIET: 'QUIET',
  NORMAL: 'NORMAL'
};

exports.SpaceSize = exports.$Enums.SpaceSize = {
  SMALL: 'SMALL',
  MEDIUM: 'MEDIUM',
  LARGE: 'LARGE'
};

exports.EquipmentCategory = exports.$Enums.EquipmentCategory = {
  NONE: 'NONE',
  FURNITURE: 'FURNITURE',
  WALL: 'WALL',
  BOTTLE: 'BOTTLE',
  BAG: 'BAG',
  STAIRS: 'STAIRS',
  FABRIC: 'FABRIC',
  STICK: 'STICK',
  OUTDOOR: 'OUTDOOR',
  CREATIVE: 'CREATIVE'
};

exports.PrimaryMuscle = exports.$Enums.PrimaryMuscle = {
  CHEST: 'CHEST',
  BACK: 'BACK',
  LEGS: 'LEGS',
  GLUTES: 'GLUTES',
  SHOULDERS: 'SHOULDERS',
  ARMS: 'ARMS',
  CORE: 'CORE',
  FULL_BODY: 'FULL_BODY',
  NECK_SHOULDER: 'NECK_SHOULDER'
};

exports.IntentType = exports.$Enums.IntentType = {
  RELAX: 'RELAX',
  STRETCH: 'STRETCH',
  MODERATE: 'MODERATE',
  STRENGTH: 'STRENGTH'
};

exports.Difficulty = exports.$Enums.Difficulty = {
  GREEN: 'GREEN',
  BLUE: 'BLUE',
  RED: 'RED'
};

exports.DurationType = exports.$Enums.DurationType = {
  TIME: 'TIME',
  REPS: 'REPS'
};

exports.SessionStatus = exports.$Enums.SessionStatus = {
  PENDING: 'PENDING',
  IN_PROGRESS: 'IN_PROGRESS',
  COMPLETED: 'COMPLETED',
  ABANDONED: 'ABANDONED'
};

exports.RarityLevel = exports.$Enums.RarityLevel = {
  COMMON: 'COMMON',
  UNCOMMON: 'UNCOMMON',
  RARE: 'RARE',
  EPIC: 'EPIC',
  LEGENDARY: 'LEGENDARY'
};

exports.DataSource = exports.$Enums.DataSource = {
  WEEKLY_TABLE: 'WEEKLY_TABLE',
  ON_THE_FLY_ESTIMATE: 'ON_THE_FLY_ESTIMATE'
};

exports.DeeplinkTargetType = exports.$Enums.DeeplinkTargetType = {
  WORKOUT_SESSION: 'WORKOUT_SESSION',
  THEME_WEEK: 'THEME_WEEK',
  SHARE_CARD: 'SHARE_CARD',
  EXERCISE: 'EXERCISE'
};

exports.Prisma.ModelName = {
  Scenario: 'Scenario',
  Equipment: 'Equipment',
  Exercise: 'Exercise',
  ExerciseScenario: 'ExerciseScenario',
  ExerciseEquipment: 'ExerciseEquipment',
  User: 'User',
  WorkoutSession: 'WorkoutSession',
  SessionExercise: 'SessionExercise',
  ShareCard: 'ShareCard',
  DailyTraining: 'DailyTraining',
  RarityTable: 'RarityTable',
  UserPreference: 'UserPreference',
  ThemeWeek: 'ThemeWeek',
  ThemeWeekParticipation: 'ThemeWeekParticipation',
  Deeplink: 'Deeplink',
  DeeplinkClick: 'DeeplinkClick'
};

/**
 * This is a stub Prisma Client that will error at runtime if called.
 */
class PrismaClient {
  constructor() {
    return new Proxy(this, {
      get(target, prop) {
        let message
        const runtime = getRuntime()
        if (runtime.isEdge) {
          message = `PrismaClient is not configured to run in ${runtime.prettyName}. In order to run Prisma Client on edge runtime, either:
- Use Prisma Accelerate: https://pris.ly/d/accelerate
- Use Driver Adapters: https://pris.ly/d/driver-adapters
`;
        } else {
          message = 'PrismaClient is unable to run in this browser environment, or has been bundled for the browser (running in `' + runtime.prettyName + '`).'
        }

        message += `
If this is unexpected, please open an issue: https://pris.ly/prisma-prisma-bug-report`

        throw new Error(message)
      }
    })
  }
}

exports.PrismaClient = PrismaClient

Object.assign(exports, Prisma)
